### PR TITLE
[WEB] Removing shared module, passing Store/ActionsProvider in context.

### PR DESCRIPTION
**Pull request by dennisoelkers**
This allows us to share singleton instances of stores and actions
throughout all components of the web interface and all plugins.

The shared bundle has been removed, because it prevents hot reloading of
shared components. Components which are reused by plugins are bundled
into plugins instead of referenced from the shared bundle. This has the
downside that plugins need rebundling when components of the web
interface are modified (because of bugfixing or enhancements) to benefit from their improvements, but it
also makes them more resilient to API changes.

### CODE REVIEW

File: graylog2-web-interface/src/injection/StoreProvider.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      UniversalSearch: () => require('stores/search/UniversalSearchStore'),
+      Session: () => require('stores/sessions/SessionStore'),
+      SystemProcessing: () => require('stores/system-processing/SystemProcessingStore'),
+      SystemShutdown: () => require('stores/system-shutdown/SystemShutdownStore'),
+      System: () => require('stores/system/SystemStore'),
+      SystemJobs: () => require('stores/systemjobs/SystemJobsStore'),
+      SystemMessages: () => require('stores/systemmessages/SystemMessagesStore'),
+      Focus: () => require('stores/tools/FocusStore'),
+      CurrentUser: () => require('stores/users/CurrentUserStore'),
+      Startpage: () => require('stores/users/StartpageStore'),
```
Comment (edmundoa) [inline graylog2-web-interface/src/injection/StoreProvider.jsx]: Could you please also add the `JournalStore`?

**Comment by edmundoa** [comment]
Other than that missing store, I guess we could also remove `graylog2-web-interface/karma.shared.js` and `graylog2-web-interface/webpack.shared.js`, now that the shared bundle is gone.

File: graylog2-web-interface/src/injection/StoreProvider.jsx (hunk 1 above)
Comment (dennisoelkers) [inline graylog2-web-interface/src/injection/StoreProvider.jsx]: I am wondering why I missed it. Was it created just recently?

**Comment by dennisoelkers** [comment]
Yes, good point! :+1:

**Comment by dennisoelkers** [comment]
Removed `webpack.shared.js` in 599d7c1

**Comment by edmundoa** [comment]
Oh, I see. Then it is good to go I think! :+1:

**Pull request merged**